File: updated_code/code/java/1876-substrings-of-size-three-with-distinct-characters.java
package com.example;

import java.util.HashSet;
import java.util.Set;

/**
 * This class is used to count the number of good substrings of length 3 in a given string.
 */
public class Solution {

    /**
     * This method is used to count the number of good substrings of length 3 in a given string.
     * 
     * @param s The input string.
     * @return The number of good substrings of length 3.
     */
    public int countGoodSubstrings(String s) {
        int count = 0;
        for (int i = 0; i < s.length() - 2; i++) {
            Set<Character> set = new HashSet<>();
            for (int j = i; j < i + 3; j++) {
                set.add(s.charAt(j));
                if (set.size() == 3) {
                    count++;
                }
            }
        }
        return count;
    }
}

File: updated_code/code/python/1876-substrings-of-size-three-with-distinct-characters.py
class Solution:
    def count_good_substrings(self, s: str) -> int:
        """
        This method counts the number of substrings of size three having distinct characters.

        Parameters:
        s (str): The input string.

        Returns:
        int: The number of substrings of size three with distinct characters.
        """
        count = 0
        for i in range(len(s) - 2):
            # Extract the substring of size 3
            substring = s[i:i+3]
            # Check if the substring has distinct characters
            if len(set(substring)) == 3:
                # If distinct, increment the count
                count += 1
        return count